src/SaveObject.ts

```diff
@@ -719,10 +730,20 @@ Error: ${e}`);
   }
 }
 
-function loadGame(saveString: string): boolean {
+async function loadGame(saveString: string): Promise<boolean> {
   createScamUpdateText();
   if (!saveString) return false;
-  saveString = decodeURIComponent(escape(atob(saveString)));
+  if (saveString.startsWith('{"CompressedSave": ')) {
```

Comment:
You're not actually using the fact that the savegame is JSON here - this is a simple magic-string check. Indeed, the save could be valid JSON containing a CompressedSave block but fail this check.

This is why I think it would be better to go with a gzipped version of the existing saveString. You could perform the same magic-string check (but with a different byte sequence), and then most of the logic would end up the same. But it would be much easier for users to look at. (Any outside tool will understand that a `.json.gz` file is just a gzipped `.json` file.)

Reply:
I mean, I could just not do the magic string check and try JSON.parse directly -  then it would be equivalent to the magic bytes approach, but more in the spirit of high level languages.

To summarize some discussion on the discord, there are several options, ordered in ascending savefile size:
- Write the compressed bytes to file with some magic bytes in front; this is optimal for performance and size
- Base64 the compressed bytes and write it to a JSON file (what this PR does)
- Compress the current savefile; this will result in a worse compression ratio, but make it easy to use external tools to edit the save

Note that the export and save formats can differ - we could write the Uint8Array to indexedDB as a blob and then only do the transformations on export/import, and I think that happens rarely enough that the performance difference is not important.

Reply:
What I was suggesting was the first option (mostly): gzip files are compressed bytes with a header (and also a minimal footer). An advantage is that the format is natively supported by CompressionStream; it's just a change in the constructor from "deflate-raw" to "gzip". I misspoke slightly when I said "gzipped version of the existing saveString," because what I actually meant was "gzipped version of the existing json" (i.e. before it goes through base64 encoding). IMO this also makes it easiest to use external tools; Windows itself can natively open .gz files.

Also agreed that we can potentially do other things for internal saves. The most obvious one would be skipping compression if it were a performance hit. Avoiding conversion to a string is another.